src/modules/previewpane/UnitTests-PreviewHandlerCommon/PreviewHandlerBaseTests.cs

```diff
@@ -0,0 +1,391 @@
+            mockPreviewControl.Setup(x => x.GetHandle())
+                .Returns(previewControlHandle);
+
+            previewHandlerControl = mockPreviewControl.Object;
+            var testPreviewHandler = new TestPreviewHandler();
+            var hwndReceived = IntPtr.Zero;
+
+            // Act
+            testPreviewHandler.GetWindow(out hwndReceived);
+
+            // Assert
+            Assert.AreEqual(hwndReceived, previewControlHandle);
+        }
+
+        [DataTestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void PreviewHandlerBase_ShouldThrowNotImplementedException_IfContextSensitiveHelpCalled(bool fEnterMode)
+        {
+            // Arrange
+            var mockPreviewControl = new Mock<IPreviewHandlerControl>();
+
+            previewHandlerControl = mockPreviewControl.Object;
+            var testPreviewHandler = new TestPreviewHandler();
+            NotImplementedException exception = null;
+
+            // Act
+            try
+            {
+                testPreviewHandler.ContextSensitiveHelp(fEnterMode);
+            }
+            catch (NotImplementedException ex)
+            {
+                exception = ex;
+            }
+
+            // Assert
+            Assert.IsNotNull(exception);
+        }
+
+        [TestMethod]
+        public void PreviewHandlerBase_ShouldReturnSite_WhenGetSiteCalled()
+        {
+            // Arrange
+            var mockPreviewControl = new Mock<IPreviewHandlerControl>();
+
+            previewHandlerControl = mockPreviewControl.Object;
+            var testPreviewHandler = new TestPreviewHandler();
+            var site = new Mock<IPreviewHandlerFrame>().Object;
+            testPreviewHandler.SetSite(site);
+            object actualSite = null;
+            var riid = Guid.Empty;
+
+            // Act
+            testPreviewHandler.GetSite(ref riid, out actualSite);
+
+            // Assert
+            Assert.AreEqual(actualSite, site);
+        }
+
+        private LOGFONT GetLogFont()
+        {
+            var logFont = new LOGFONT();
+            logFont.LfHeight = 12;
+            logFont.LfWidth = 0;
+            logFont.LfEscapement = 0;
+            logFont.LfWeight = 400; // FW_NORMAL
+            logFont.LfItalic = Convert.ToByte(false);
+            logFont.LfUnderline = Convert.ToByte(false);
+            logFont.LfStrikeOut = Convert.ToByte(0);
+            logFont.LfCharSet = Convert.ToByte(0); // ANSI_CHARSET
+            logFont.LfOutPrecision = Convert.ToByte(0); // OUT_DEFAULT_PRECIS
+            logFont.LfClipPrecision = Convert.ToByte(0);
+            logFont.LfQuality = Convert.ToByte(0);
+            logFont.LfPitchAndFamily = Convert.ToByte(0);
+            logFont.LfFaceName = "valid-font";
+
+            return logFont;
+        }
+
+        private RECT GetRectangle(int left, int top, int right, int bottom)
+        {
+            var rect = new RECT();

```

Comment:
Doesn't this have a constructor that takes Left, Right, Top, and Bottom as arguments?  This way you could leave the 'readonly' field on as per Divyansh's comments.

Reply:
I think you[ are referring to `Rectangle Struct`](https://docs.microsoft.com/en-us/dotnet/api/system.drawing.rectangle.fromltrb?view=netframework-4.8#System_Drawing_Rectangle_FromLTRB_System_Int32_System_Int32_System_Int32_System_Int32_) in `System.Drawing`. The defined `RECT` Struct is to get the rectangle [provided to `IPreviewHandler`](https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-ipreviewhandler-setrect) in [`RECT structure`](https://docs.microsoft.com/en-us/windows/win32/api/windef/ns-windef-rect) and convert it to `System.Drawing Rectangle`.